cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/RackAwareDistributionGoal.java

```diff
@@ -357,8 +375,9 @@ private void ensureRackAwareDistribution(ClusterModel clusterModel, Optimization
       _numAliveRacksAllowedReplicaMoves = clusterModel.numAliveRacksAllowedReplicaMoves(optimizationOptions);
       if (_numAliveRacksAllowedReplicaMoves == 0) {
         // Handle the case when all alive racks are excluded from replica moves.
-        throw new OptimizationFailureException("All alive racks are excluded from replica moves.",
-                                               String.format("Add at least %d brokers to different racks.", clusterModel.maxReplicationFactor()));
+        ProvisionRecommendation recommendation = new ProvisionRecommendation.Builder(ProvisionStatus.UNDER_PROVISIONED)
+            .numBrokers(clusterModel.maxReplicationFactor()).build();
+        throw new OptimizationFailureException("All alive racks are excluded from replica moves.", recommendation);
```

Comment:
IIUC, the patch would only prompt users to add X brokers. But adding all X brokers to the same rack would not be enough, they must add brokers on different racks. Maybe we can add more instructions to make it more clear?
```suggestion
        throw new OptimizationFailureException("All alive racks are excluded from replica moves, please add more brokers to different racks.", recommendation);
```

Reply:
Thanks for pointing this out -- the recommendation should indeed ask for `Adding racks with brokers` rather than `Adding brokers`. Fixed the relevant provision recommendation.